File: exercises/D702/solution.pas
var
    st: array[1..1000] of longint;
    n, i, dummy, size: longint;
    s: string;
begin
    size := 0;
    readln(n);
    for i := 1 to n do
    begin
        readln(s);
        if (copy(s, 1, 4) = 'PUSH') then
        begin
            inc(size);
            val(copy(s, 6, 10), st[size], dummy)
        end
        else if (s = 'TOP') then
        begin
            if (size = 0) then
                writeln('Empty')
            else
                writeln(st[size])
        end
        else if (s = 'POP') then
        begin
            if (size = 0) then
                writeln('Cannot pop')
            else
                dec(size)
        end
        else
            writeln(size)
    end
end.

File: exercises/D702/solution.py
n = int(input())
size = 0
st = []
for _ in range(n):
    args = input().split()
    if args[0] == 'PUSH':
        st.append(args[1])
        size += 1
    elif args[0] == 'TOP':
        if size == 0:
            print('Empty')
        else:
            print(st[size - 1])
    elif args[0] == 'POP':
        if size == 0:
            print('Cannot pop')
        else:
            st.pop(-1)
            size -= 1
    else:
        print(size)
